File: examples/java7-example/src/main/java/nz/bradcampbell/paperparcel/java7example/DateTypeAdapter.java
package nz.bradcampbell.paperparcel.java7example;

import android.os.Parcel;

import nz.bradcampbell.paperparcel.TypeAdapter;
import org.jetbrains.annotations.NotNull;

import java.util.Date;

public class DateTypeAdapter implements TypeAdapter<Date> {
  @NotNull
  @Override
  public Date readFromParcel(@NotNull Parcel inParcel) {
    return new Date(inParcel.readLong());
  }

  @Override
  public void writeToParcel(@NotNull Date value, @NotNull Parcel outParcel, int flags) {
    outParcel.writeLong(value.getTime());
  }
}

File: examples/kotlin-example/src/main/kotlin/nz/bradcampbell/paperparcel/kotlinexample/DateTypeAdapter.kt
package nz.bradcampbell.paperparcel.kotlinexample

import android.os.Parcel
import nz.bradcampbell.paperparcel.typeadapters.base.AbstractAdapter
import java.util.*

object DateTypeAdapter : AbstractAdapter<Date>() {
  override fun write(value: Date, dest: Parcel, flags: Int) {
    dest.writeLong(value.time);
  }

  override fun read(source: Parcel): Date {
    return Date(source.readLong())
  }
}
